### PR TITLE
fix bug about can't skip commits base on base branch

### LINKED ISSUE #11536: [Bug] commits history comment on PR show unrelated after merge
a merge will display all comits

what I expect: only chow the merge commit

demo: https://gitea.com/gitea/go-sdk/pulls/273#issuecomment-113383

function introduced in #11167

### PR DESCRIPTION
**Pull request by a1012112796**
fix bug about can't skip commits base on base branch. I know it's not a good solution, but I have no other way now.

test with demo:

image

fix #11536

### CODE REVIEW

**Comment by a1012112796** [comment]
Please review, Thanks :)

**Comment by 6543** [comment]
O nice - will check it out :tada:

File: modules/git/commit.go [hunk 1]
```diff
@@ -482,6 +482,16 @@ func (c *Commit) GetBranchName() (string, error) {
 	return strings.SplitN(strings.TrimSpace(data), "~", 2)[0], nil
 }
 
+// LoadBranchName load branch name for commit
+func (c *Commit) LoadBranchName() (err error) {
+	if c.Branch != "" {
```
Comment (lafriks) [inline modules/git/commit.go]:
```suggestion
	if len(c.Branch) != 0 {
```

File: models/issue_comment.go [hunk 2]
```diff
@@ -1124,12 +1124,11 @@ func getCommitIDsFromRepo(repo *Repository, oldCommitID, newCommitID, baseBranch
...
 	}
 
-	oldCommitBranch, err := oldCommit.GetBranchName()
-	if err != nil {
+	if err = oldCommit.LoadBranchName(); err != nil {
 		return nil, false, err
 	}
 
-	if oldCommitBranch == "" {
+	if oldCommit.Branch == "" {
```
Comment (lafriks) [inline models/issue_comment.go]:
```suggestion
	if len(oldCommit.Branch) == 0{
```

**Comment by 6543** [comment]
I'm actualy just thinking how i can reproduce the bug to test your patch 🤔

**Comment by a1012112796** [comment]
> I'm actualy just thinking how i can reproduce the bug to test your patch 🤔

Maybe you can copy go-sdk to your test env with“New Migration”.

**Comment by 6543** [comment]
finaly had time and could reproduce it - this works perfect :)

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
big thanks for fixing this anoying bug 👍

**Review by mrsdizzie — APPROVED** [review]

**Comment by zeripath** [comment]
make lg-tm work

**Pull request merged**